opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/async/AsyncCustomAutoConfiguration.java

```diff
@@ -0,0 +1,38 @@
+package io.opentracing.contrib.spring.cloud.async;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.AsyncConfigurer;
+import org.springframework.scheduling.annotation.EnableAsync;
+
+/**
+ * @author kamesh
+ */
+@Configuration
+@ConditionalOnBean(AsyncConfigurer.class)
+@AutoConfigureBefore(AsyncDefaultAutoConfiguration.class)
+//TODO when Scheduling is added we need to do @AutoConfigurationAfter here
+public class AsyncCustomAutoConfiguration implements BeanPostProcessor {
```

Comment:
this has to be added to spring factories file. (located in resources)

Reply:
sorry missed in previous commit, added to new commit

Reply:
Could you please add class level comment what it does, what it traces?